### DISCUSSION TITLE
start dishwasher program and auth problems

### DISCUSSION THREAD

**Discussion by Schmello** [alexkn/node-red-contrib-homeconnect#25] [Q&A] [answered]
Is it possible to get an example of how to turn on / select program / start my dishwasher?
I can't get this going

In addition, I have to restart the Auth after each deploy.

**Comment by alexkn** [comment]
I moved the issue to discussions.
The issue tracker is for bug reports, not for questions.

Please describe exactly what you already tried. 
Maybe then I can help you.

**Comment by Schmello** [comment]
First :) 
If I change something on the node and click on deploy, I always have to re-authenticate myself.
But if I change something and remove the callback URL in the Auth Node and then click on deploy, I don't have to re-authenticate.

I got my HAID out and the command to switch on my dishwasher.
But unfortunately this doesn't work because I don't have the information that needs to be on the body.
What has to go in there?
![geschir](https://user-images.githubusercontent.com/22894811/105384060-149e0880-5c12-11eb-9017-ebefa05c1804.jpg)

Is it possible to send the following commands at once via a node?
ON / program / start?
Or does each command have to be sent one after the other?

In the worst case, can the dishwasher be switched on even though the door is open? :)

**Comment by alexkn** [comment]
I don't know whats wrong with the re-authenticate.
I never see such problem.

set_settings action needs a body, see https://apiclient.home-connect.com/#/settings/set_setting
But I think you don't need that, you can use start_program directly.
You have to active remote start directly at the dishwasher.
see https://apiclient.home-connect.com/#/programs/start_program for details.

**Comment by Schmello** [comment]
Hello and thanks for the quick reply.

```
Switching on and off works now. over
Settings
set Setting
Haid (XXXXXX)
Setting Key (BSH.Common.Setting.PowerState)
Body
 ({
  "data": {
    "key": "BSH.Common.Setting.PowerState",
    "value": "BSH.Common.EnumType.PowerState.Off",
    "type": "BSH.Common.EnumType.PowerState",
    "constraints": {
      "allowedvalues": [
        "BSH.Common.EnumType.PowerState.On",
        "BSH.Common.EnumType.PowerState.Off"
      ]
    }
  }
})
```

When i try this after start the dishwasher
Tag - Program
action - start_program
Haid - (XXXXXXX)
Body - ? 

I think that I have to set the program first and then the program options before I can start, right?
Output Program Option
`{"data":{"options":[{"key":"BSH.Common.Option.StartInRelative","value":0,"unit":"seconds"},{"key":"Dishcare.Dishwasher.Option.ExtraDry","value":false},{"key":"Dishcare.Dishwasher.Option.IntensivZone","value":false},{"key":"Dishcare.Dishwasher.Option.VarioSpeedPlus","value":false},{"key":"Dishcare.Dishwasher.Option.SilenceOnDemand","value":false}]}}`

Output Programs
`{"data":{"programs":[{"key":"Dishcare.Dishwasher.Program.Intensiv70","constraints":{"execution":"selectandstart"}},{"key":"Dishcare.Dishwasher.Program.Auto2","constraints":{"execution":"selectandstart"}},{"key":"Dishcare.Dishwasher.Program.Eco50","constraints":{"execution":"selectandstart"}},{"key":"Dishcare.Dishwasher.Program.Glas40","constraints":{"execution":"selectandstart"}},{"key":"Dishcare.Dishwasher.Program.Quick45","constraints":{"execution":"selectandstart"}},{"key":"Dishcare.Dishwasher.Program.NightWash","constraints":{"execution":"selectandstart"}},{"key":"Dishcare.Dishwasher.Program.Kurz60","constraints":{"execution":"selectandstart"}},{"key":"Dishcare.Dishwasher.Program.MachineCare","constraints":{"execution":"selectandstart"}}]}}`

**Comment by alexkn** [comment, marked as answer]
This should work:
```
{
    "data": {
        "key": "Dishcare.Dishwasher.Program.Auto2",
        "options": [
            {
                "key": "BSH.Common.Option.StartInRelative",
                "value": 0,
                "unit": "seconds"
            }
        ]
    }
}
```

**Comment by Schmello** [comment]
Thank you it works. :)
At first it hadn't worked.
With Value, the space had to be removed before the 0 :)
`"value":0,`
Is the value a time delay in seconds?

How do I get the program options?
`{"data":{"options":[{"key":"BSH.Common.Option.StartInRelative","value":0,"unit":"seconds"},{"key":"Dishcare.Dishwasher.Option.ExtraDry","value":false},{"key":"Dishcare.Dishwasher.Option.IntensivZone","value":false},{"key":"Dishcare.Dishwasher.Option.VarioSpeedPlus","value":false},{"key":"Dishcare.Dishwasher.Option.SilenceOnDemand","value":false}]}}`

**Reply by alexkn** [reply]
I am not able to give examples for everything.
Please read the Home-Connect docs.
https://apiclient.home-connect.com/

**Comment by Schmello** [comment]
No problem :) Thanks for your help.

**Comment by Schmello** [comment]
Can you tell me if I get the information for the program options in the body from the start? Or do I have to set that separately?
I've tried the following, it doesn't work.

```
{
    "data": {
        "key": "Dishcare.Dishwasher.Program.Auto2",
        "options": [
            {
                "key": "BSH.Common.Option.StartInRelative",
                "value": 0,
                "unit": "seconds"
            },
{
                "key": "Dishcare.Dishwasher.Option.ExtraDry",
                "value": true,
            }
        ]
    }
}
```
